### PR TITLE
[MM-59338] Migrate tooltips of "components/menu/menu.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Camillarhi**
#### Summary

Migrate tooltips to use WithTooltip in "components/menu/menu.tsx"

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27554
Jira https://mattermost.atlassian.net/browse/MM-59338

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/menu/menu.tsx [hunk 1]
```diff
@@ -203,21 +201,13 @@ export function Menu(props: Props) {
...
                     placement={props?.menuButtonTooltip?.placement ?? 'top'}
-                    overlay={
-                        <Tooltip
-                            id={props.menuButtonTooltip.id}
-                            className={props.menuButtonTooltip?.class ?? ''}
-                        >
-                            {props.menuButtonTooltip.text}
-                        </Tooltip>
-                    }
-                    disabled={isMenuOpen}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/menu/menu.tsx:217]: We didnt add disabled back here

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Can you also check the failing CI

File: webapp/channels/src/components/menu/menu.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/menu/menu.tsx:217]: @larkox can you please verify the disabled prop behavior do we need that?

Comment (Camillarhi) [reply webapp/channels/src/components/menu/menu.tsx:217]: @M-ZubairAhmed  I have added the disabled

Comment (M-ZubairAhmed) [reply webapp/channels/src/components/menu/menu.tsx:217]: @Camillarhi CI is still failing can you check once again

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Can you check the failing CI

**Comment by M-ZubairAhmed** [comment]
> I am not much fond of increasing even more the `withTooltip` interface, but I guess that `disabled` is somehow nice to have. One part of me wants to keep it as:
> 
> ```
> foo = disabled ? <>{children}</> : <WithTooltip ...>{children}</WithTooltip>;
> ```
> 
> But the more I think about it, the more I see the value of the prop.

Agreed may be in the new implementation we just to this when we get the disabled prop

**Comment by Camillarhi** [comment]
> Can you check the failing CI

The failing tests doesn't seem to come from the `menu.tsx` file

**Comment by M-ZubairAhmed** [comment]
> > Can you check the failing CI
> 
> The failing tests doesn't seem to come from the `menu.tsx` file

The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.

**Comment by Camillarhi** [comment]
> > > Can you check the failing CI
> > 
> > 
> > The failing tests doesn't seem to come from the `menu.tsx` file
> 
> The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.

How do I proceed?

**Comment by Camillarhi** [comment]
> > > > > Can you check the failing CI
> > > > 
> > > > 
> > > > The failing tests doesn't seem to come from the `menu.tsx` file
> > > 
> > > 
> > > The component menu is used in other places and because the structure of the menu changed and hence the snapshots where this component is used have also changed. So it is because of the change itself.
> > 
> > 
> > How do I proceed?
> 
> You can update the snapshots with `npm run test:updatesnapshot`

I have updated the snapshots

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
~@Camillarhi Oops sorry it looks like some more are still failing~

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**